src/mlpack/methods/decision_tree/decision_tree_regressor_impl.hpp

```diff
@@ -811,36 +811,49 @@ double DecisionTreeRegressor<FitnessFunction,
       responses.cols(begin, begin + count - 1),
       UseWeights ? weights.subvec(begin, begin + count - 1) : weights);
   size_t bestDim = data.n_rows; // This means "no split".
+  const size_t nDims = dimensionSelector.NumDimensions();
 
   if (maximumDepth != 1)
   {
-    for (size_t i = dimensionSelector.Begin(); i != dimensionSelector.End();
-         i = dimensionSelector.Next())
+    #pragma omp parallel
     {
-      const double dimGain = NumericSplitType<FitnessFunction>::template
-          SplitIfBetter<UseWeights>(bestGain,
-                                    data.cols(begin, begin + count - 1).row(i),
-                                    responses.cols(begin, begin + count - 1),
-                                    UseWeights ?
-                                        weights.cols(begin, begin + count - 1) :
-                                        weights,
-                                    minimumLeafSize,
-                                    minimumGainSplit,
-                                    splitPointOrPrediction,
-                                    *this,
-                                    fitnessFunction);
-
-      // If the splitter did not report that it improved, then move to the next
-      // dimension.
-      if (dimGain == DBL_MAX)
-        continue;
-
-      bestDim = i;
-      bestGain = dimGain;
+      double localBestGain = 0;
+      size_t localBestDim = 0;
+      #pragma omp for reduction( max : bestGain)
```

Comment:
Maybe I missed it, but did you already run some benchmark against the parallelization code?